conformance/lang/expressions/shift-expression/left_shift.balt

```diff
@@ -0,0 +1,5522 @@
+    //                After applying mask, In binary: 000000, In decimal: 0
+    // Result = In binary: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00111111
+    //        = In decimal: (63)
+
+    io:println(a << e); // @output -9223372036854775808
+    // Explanation:
+    // Value to be shifted = Equivalent binary representation : 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111
+    // Shift amount = Before applying mask, In binary: 11111111 11111111 11111111 11111111 11111111 11111111 11111111 10111111, In decimal: -65
+    //                After applying mask, In binary: 111111, In decimal: 63
+    // Result = In binary: 10000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+    //        = In decimal: (-9223372036854775808)
+
+    io:println(a << f); // @output -1
+    // Explanation:
+    // Value to be shifted = Equivalent binary representation : 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111
+    // Shift amount = Before applying mask, In binary: 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11000000, In decimal: -64
+    //                After applying mask, In binary: 000000, In decimal: 0
+    // Result = In binary: 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111
+    //        = In decimal: (-1)
+
+    io:println(a << a); // @output -9223372036854775808
+    // Explanation:
+    // Value to be shifted = Equivalent binary representation : 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111
+    // Shift amount = Before applying mask, In binary: 11111111 11111111 11111111 11111111 11111111 11111111 11111111 11111111, In decimal: -1
+    //                After applying mask, In binary: 111111, In decimal: 63
+    // Result = In binary: 10000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+    //        = In decimal: (-9223372036854775808)
+
+    io:println(e << c); // @output -65
+    // Explanation:
+    // Value to be shifted = Equivalent binary representation : 11111111 11111111 11111111 11111111 11111111 11111111 11111111 10111111
+    // Shift amount = Before applying mask, In binary: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 01000000, In decimal: 64
+    //                After applying mask, In binary: 000000, In decimal: 0
+    // Result = In binary: 11111111 11111111 11111111 11111111 11111111 11111111 11111111 10111111
+    //        = In decimal: (-65)
+}
+
+Test-Case: error
+Description: Test resultant type of the '<<' expression with 'int' operands.
+Labels: decimal, float, int, shift-expr, unary-minus, var
+
+function errorFunction() {
+    int a = 128;
+    int b = 7;
+    int c = -120;
+    int d = -9223372036854775794;
+
+    var x1 = a << b;
+    decimal _ = x1; // @error expected 'decimal' but found 'int'
+    var x2 = a << d;
+    float _ = x2; // @error expected 'float' but found 'int'
```

Comment:
without mentioning the error itself; say that the expressions' static type with operands of type bla bla was int, but expecting ** another type**. Check all cases where applicable

Reply:
Since this error msg also explains about the reason for the error, do we need to update it? WDYT

Reply:
assignment error is fine. but I was referring to why it became that type as I mentioned in https://github.com/ballerina-platform/ballerina-spec/pull/953#discussion_r727200078